packages/server/src/fhir/repo.ts

```diff
@@ -2168,18 +2171,29 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
     callback: (client: PoolClient) => Promise<TResult>,
     options?: { serializable: boolean }
   ): Promise<TResult> {
-    try {
-      const client = await this.beginTransaction(options?.serializable ? 'SERIALIZABLE' : undefined);
-      const result = await callback(client);
-      await this.commitTransaction();
-      return result;
-    } catch (err) {
-      const operationOutcomeError = normalizeDatabaseError(err);
-      await this.rollbackTransaction(operationOutcomeError);
-      throw operationOutcomeError;
-    } finally {
-      this.endTransaction();
+    let error: any;
+    for (let i = 0; i < transactionAttempts; i++) {
+      try {
+        const client = await this.beginTransaction(options?.serializable ? 'SERIALIZABLE' : undefined);
+        const result = await callback(client);
+        await this.commitTransaction();
+        return result;
+      } catch (err) {
+        const operationOutcomeError = normalizeDatabaseError(err);
+        // Assigning here and throwing below is necessary to satisfy TypeScript
+        error = err;
```

Comment:
Small nit, could cast to `OperationOutcomeError | Error`:
```ts
let error:  OperationOutcomeError | Error;
```

...

```ts
error = err as OperationOutcomeError | Error;
```

Then pass it into `this.isRetryableTransactionError` and make signature:

```ts
  private isRetryableTransactionError(err: OperationOutcomeError | Error): boolean {
```

Reply:
Since `isRetryableTransactionError()` tries accessing `err.code`, it requires `any` or perhaps an interface type: that property doesn't exist on either `OperationOutcomeError` or generic `Error`

Reply:
Oh good point